examples/async/fastapi_example.py

```diff
@@ -1,13 +1,61 @@
+import logging
+from contextlib import asynccontextmanager
+
+import aiohttp
 import async_module
 import fastapi
+from aiohttp import client_exceptions
+from hamilton_sdk import adapters
 
-from hamilton import base
 from hamilton.experimental import h_async
 
-app = fastapi.FastAPI()
+logger = logging.getLogger(__name__)
 
 # can instantiate a driver once for the life of the app:
-dr = h_async.AsyncDriver({}, async_module, result_builder=base.DictResult())
+dr = None
+
+
+async def _tracking_server_running():
+    """Quickly tells if the tracking server is up and running"""
+    async with aiohttp.ClientSession() as session:
+        try:
+            async with session.get("http://localhost:8241/api/v1/ping") as response:
+                if response.status == 200:
+                    return True
+                else:
+                    return False
+        except client_exceptions.ClientConnectionError:
+            return False
+
+
+@asynccontextmanager
+async def lifespan(app: fastapi.FastAPI):
```

Comment:
It would be nice to have a second endpoint to compare latency/penalty for the tracker.